### PR TITLE
Upgrade printing dependencies

### PR DESCRIPTION
**Pull request by ari**
6 years of JasperReports upgrades

### CODE REVIEW

**Comment by ari** [comment]
> George, could you test printing with this upgrade. Backgrounds, multi page backgrounds, and complicated reports like certificates.

Do we still have any reports with graphs?

**Comment by ari** [comment]
George, where are we with this?

**Review by ArtyomKravchenko — DISMISSED** [review]

**Comment by GFilipovich** [comment]
Created https://ish-group.atlassian.net/browse/OD-16217 with issues found in this branch.

File: server-api/src/main/groovy/ish/oncourse/server/api/cxf/ObjectMapperContextResolver.groovy [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Provider;
+
+
+@Provider
+@Produces(MediaType.APPLICATION_JSON)
+class ObjectMapperContextResolver implements ContextResolver<ObjectMapper> {
```
Comment (ari) [inline server-api/src/main/groovy/ish/oncourse/server/api/cxf/ObjectMapperContextResolver.groovy:29]: Can we have some javadocs here to explain what the purpose of this class is.

**Comment by ari** [comment]
@yuryyas Your commits in this branch are a bit confusing. Are they meant to be here?

**Comment by KeyrisXdSnow** [comment]
To solve problem with border in report we can try use 4 settings preferences:
net.sf.jasperreports.style.box.bottom.pen.lineWidth
net.sf.jasperreports.style.box.top.pen.lineWidth
net.sf.jasperreports.style.box.left.pen.lineWidth
net.sf.jasperreports.style.box.right.pen.lineWidth

Just need set up them in jasperreports.properties

**Pull request closed without merging**